### PR TITLE
Rhidp 3835 home page document how users can customize the home page

### PR DESCRIPTION
**Pull request by jmagak**
**IMPORTANT: Do Not Merge - To be merged by Docs Team Only**

**Version(s):** 1.3 

**Issue:** https://issues.redhat.com/browse/RHIDP-3835

**Link to docs preview:**

**Reviews:**
- [ ] SME: @christoph-jerolimov 
- [ ] QE: @ mention assignee
- [ ] Docs review: @themr0c 
- [ ] Additional review: @mention assignee (by writer)

**Additional information:**

### CODE REVIEW

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 1]
```diff
@@ -23,22 +39,362 @@ To access the data from the JSON files, complete the following step:
...
-    # customize developer hub instance
-    '/developer-hub':
-      target: <DOMAIN_URL> # i.e https://raw.githubusercontent.com/
-      pathRewrite:
-        '^/api/proxy/developer-hub': <path to json file> # i.e /janus-idp/backstage-showcase/main/packages/app/public/homepage/data.json
-      changeOrigin: true
-      secure: true
-      # Change to "false" in case of using self hosted cluster with a self-signed certificate
-      headers:
-	<HEADER_KEY>: <HEADER_VALUE> # optional and can be passed as needed i.e Authorization can be passed for private GitHub repo and PRIVATE-TOKEN can be passed for private GitLab repo
```
Comment (jerolimov) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:38]: As discussed in google meet, we should restore that section.

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 2]
```diff
@@ -6,7 +6,23 @@ To access the Home page in {product}, the base URL must include the `/developer-
...
+* Customize and extend the default Home page layout with additional cards that appear based on the plugins you have installed and enabled. 
+* Change quick access links.  
+* Add, reorganize, and remove the following available cards:
++
+** Search bar
+** Quick access
+** Headline
+** Markdown
+** Placeholder
+** Catalog starred entities
```
Comment (jerolimov) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:23]: As discussed in Google meet, I think this is a good intro into the section 4 at all.

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 3]
```diff
@@ -6,7 +6,23 @@ To access the Home page in {product}, the base URL must include the `/developer-
 * JSON files hosted on GitHub or GitLab.
 * A dedicated service that provides the Home page data in JSON format using an API.
```
Comment (jerolimov) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:7]: This intro text should go as an intro text of the new 4.2.

**Review by jerolimov — CHANGES_REQUESTED** [review]
I would suggest some changes as we discussed on Google Meet.

Sharing this here so that other reviewers are aware. :)

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 4]
```diff
@@ -1,12 +1,394 @@
...
+          importName: CatalogStarredEntitiesCard
+          config:
+            layouts:
+              xl: { w: 5, h: 4, x: 7 }
+              lg: { w: 5, h: 4, x: 7 }
+              md: { w: 5, h: 4, x: 7 }
+              sm: { w: 12, h: 4 }
+              xs: { w: 12, h: 4 }
+              xxs: { w: 12, h: 4 }
+----
```
Comment (jerolimov) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:59]: I would move that section under 

```
== Customizing the Home page cards

Administrators can change the fixed height of cards that are in a 12-column grid.
```

as its part of the customizing.

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 5]
```diff
@@ -1,12 +1,394 @@
...
+        - mountPoint: home.page/cards
+          importName: SearchBar
+          config:
+            layouts:
+              xl: { w: 10, h: 1, x: 1 }
+              lg: { w: 10, h: 1, x: 1 }
+              md: { w: 10, h: 1, x: 1 }
+              sm: { w: 10, h: 1, x: 1 }
+              xs: { w: 12, h: 1 }
+              xxs: { w: 12, h: 1 }
```
Comment (jerolimov) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:194]: This is the SearchBar example, please replace it with:

```yaml
dynamicPlugins:
  frontend:
    janus-idp.backstage-plugin-dynamic-home-page:
      mountPoints:
        - mountPoint: home.page/cards
          importName: MarkdownCard
          config:
            layouts:
              xl: { w: 6, h: 4 }
              lg: { w: 6, h: 4 }
              md: { w: 6, h: 4 }
              sm: { w: 6, h: 4 }
              xs: { w: 6, h: 4 }
              xxs: { w: 6, h: 4 }
            props:
              title: Company links
              content: |
                ### RHDH
                * [Website](https://developers.redhat.com/rhdh/overview)
                * [Documentation](https://docs.redhat.com/en/documentation/red_hat_developer_hub/)
                * [GitHub Showcase](https://github.com/janus-idp/backstage-showcase)
                * [GitHub Plugins](https://github.com/janus-idp/backstage-plugins)
        - mountPoint: home.page/cards
          importName: Markdown
          config:
            layouts:
              xl: { w: 6, h: 4, x: 6 }
              lg: { w: 6, h: 4, x: 6 }
              md: { w: 6, h: 4, x: 6 }
              sm: { w: 6, h: 4, x: 6 }
              xs: { w: 6, h: 4, x: 6 }
              xxs: { w: 6, h: 4, x: 6 }
            props:
              title: Important company links
              content: |
                ### RHDH
                * [Website](https://developers.redhat.com/rhdh/overview)
                * [Documentation](https://docs.redhat.com/en/documentation/red_hat_developer_hub/)
                * [GitHub Showcase](https://github.com/janus-idp/backstage-showcase)
                * [GitHub Plugins](https://github.com/janus-idp/backstage-plugins)
```

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 6]
```diff
@@ -1,12 +1,394 @@
...
```
Comment (jerolimov) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:216]: This is the SearchBar example, please replace it with:

```yaml
dynamicPlugins:
  frontend:
    janus-idp.backstage-plugin-dynamic-home-page:
      mountPoints:
        - mountPoint: home.page/cards
          importName: Placeholder
          config:
            layouts:
              xl: { w: 1, h: 1 }
              lg: { w: 1, h: 1 }
              md: { w: 1, h: 1 }
              sm: { w: 1, h: 1 }
              xs: { w: 1, h: 1 }
              xxs: { w: 1, h: 1 }
            props:
              showBorder: true
              debugContent: '1'
```

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 7]
```diff
@@ -1,12 +1,394 @@
...
```
Comment (jerolimov) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:238]: This is also the searchbar example, please replace it with:

```yaml
dynamicPlugins:
  frontend:
    janus-idp.backstage-plugin-dynamic-home-page:
      mountPoints:
        - mountPoint: home.page/cards
          importName: CatalogStarredEntitiesCard
```

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 8]
```diff
@@ -1,12 +1,394 @@
...
```
Comment (jerolimov) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:260]: Please update to:

```yaml
dynamicPlugins:
  frontend:
    janus-idp.backstage-plugin-dynamic-home-page:
      mountPoints:
        - mountPoint: home.page/cards
          importName: FeaturedDocsCard
```

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 9]
```diff
@@ -38,7 +420,8 @@ proxy:
 	<HEADER_KEY>: <HEADER_VALUE> # optional and can be passed as needed i.e Authorization can be passed for private GitHub repo and PRIVATE-TOKEN can be passed for private GitLab repo
 ----
 
-== Using a dedicated service to provide data to the Learning Paths
+
+=== Using a dedicated service to provide data to the Home page
```
Comment (jerolimov) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:424]:
```suggestion
=== Using a dedicated service to provide data to the Quick access card
```

File: modules/getting-started/proc-customize-rhdh-homepage.adoc (hunk 7 above)
Comment (jmagak) [reply modules/getting-started/proc-customize-rhdh-homepage.adoc:238]: Hi @jerolimov, for this yaml and the one for the featured docs, does it mean that we are leaving out the layouts?

**Comment by themr0c** [comment]
This pull request requires rebase. Is it still relevant?

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 10]
```diff
@@ -1,12 +1,405 @@
...
++
+** Search bar
+** Quick access
+** Headline
+** Markdown
+** Placeholder
+** Catalog starred entities
+** Featured docs
+
+== Customizing the Home page cards
```
Comment (themr0c) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:18]: This should go in a dedicated procedure file.

File: modules/getting-started/proc-customize-rhdh-homepage.adoc [hunk 11]
```diff
@@ -1,12 +1,405 @@
...
+dynamicPlugins:
+  frontend:
+    janus-idp.backstage-plugin-dynamic-home-page:
+      mountPoints:
+        - mountPoint: home.page/cards
+          importName: FeaturedDocsCard
+----
+--
+
+=== Defining the layout of the {product} Home page
```
Comment (themr0c) [inline modules/getting-started/proc-customize-rhdh-homepage.adoc:276]: This should be in a dedicated procedure file.

**Review by christoph-jerolimov — APPROVED** [review]
Lgtm!

**Review by teknaS47 — APPROVED** [review]

File: assemblies/assembly-customizing-the-quick-access-card-in-rhdh.adoc [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+ifdef::context[:parent-context-of-customizing-the-quick-access-card-in-rhdh: {context}]
+
+:_mod-docs-content-type: ASSEMBLY
+
+ifndef::context[]
+[id="customizing-the-quick-access-card-in-rhdh"]
+endif::[]
+ifdef::context[]
+[id="customizing-the-quick-access-card-in-rhdh_{context}"]
+endif::[]
```
Comment (themr0c) [inline assemblies/assembly-customizing-the-quick-access-card-in-rhdh.adoc:13]: You can safely simplify the metadata:

```suggestion
:_mod-docs-content-type: ASSEMBLY

[id="customizing-the-quick-access-card-in-rhdh"]
```

File: modules/getting-started/proc-defining-the-layout-of-the-product-home-page.adoc [hunk 13]
```diff
@@ -0,0 +1,122 @@
...
+= Defining the layout of the {product} Home page
+
+.Prerequisites
+* Include the following optimal parameters in each of your breakpoints:
+** width (w)
+** height (h)
+** position (x and y)
+
+.Procedure
+* Use the full space on smaller windows and half of the space on larger windows as follows:
```
Comment (themr0c) [inline modules/getting-started/proc-defining-the-layout-of-the-product-home-page.adoc:15]:
```suggestion
. Use the full space on smaller windows and half of the space on larger windows as follows:
```

File: modules/getting-started/proc-defining-the-layout-of-the-product-home-page.adoc [hunk 14]
```diff
@@ -0,0 +1,122 @@
...
+              md: { w: 6, h: 2 }
+              sm: { w: 12, h: 2 }
+              xs: { w: 12, h: 2 }
+              xxs: { w: 12, h: 2 }
+            props:
+              showBorder: true
+              debugContent: a placeholder
+----
+
+* Show the cards side by side by defining the `x` parameter as follows:
```
Comment (themr0c) [inline modules/getting-started/proc-defining-the-layout-of-the-product-home-page.adoc:38]:
```suggestion
. Show the cards side by side by defining the `x` parameter as follows:
```

File: modules/getting-started/proc-using-hosted-json-files-to-provide-data-to-the-quick-access-card.adoc [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+.Prerequisites
+
+* You have installed {product} by using either the Operator or Helm chart.
+See xref:{installing-on-ocp-book-url}#assembly-install-rhdh-ocp[{installing-on-ocp-book-title}].
+
+.Procedure
+
+To access the data from the JSON files:
+
+* Add the following code to the `app-config.yaml` file:
```
Comment (themr0c) [inline modules/getting-started/proc-using-hosted-json-files-to-provide-data-to-the-quick-access-card.adoc:18]:
```suggestion
* To access the data from the JSON files, add the following code to the {product-short} `app-config.yaml` conbfiguration file:
```

**Comment by themr0c** [comment]
Nice modularization. Only some nitpicks.

**Review by themr0c — APPROVED** [review]

**Pull request merged**